lib/core/test/integration/Test/Integration/Scenario/API/Wallets.hs

```diff
@@ -1460,6 +1464,127 @@ spec = do
             r <- request @ApiWallet ctx (method, endpoint) Default Empty
             expectResponseCode @IO HTTP.status405 r
             expectErrorMessage errMsg405 r
+
+    it "WALLETS_UTXO_01 - Wallet's inactivity is reflected in utxo" $ \ctx -> do
+        w <- emptyWallet ctx
+        rStat <- request @ApiUtxoStatistics ctx (getWalletUtxoEp w) Default Empty
+        expectResponseCode @IO HTTP.status200 rStat
+        expectWalletUTxO [] (snd rStat)
+
+    it "WALLETS_UTXO_02 - Sending and receiving funds updates wallet's utxo." $ \ctx -> do
+        wSrc <- fixtureWallet ctx
+        wDest <- emptyWallet ctx
+
+        --send funds
+        addrs <- listAddresses ctx wDest
+        let destination = (addrs !! 1) ^. #id
+        let coins = [13::Natural, 43, 66, 101, 1339]
+        let matrix = zip coins [1..]
+        forM_ matrix $ \(c, alreadyAbsorbed) -> do
+            let payload = Json [json|{
+                    "payments": [{
+                        "address": #{destination},
+                        "amount": {
+                            "quantity": #{c},
+                            "unit": "lovelace"
+                        }
+                    }],
+                    "passphrase": "cardano-wallet"
+                }|]
+            rTrans <- request @(ApiTransaction t) ctx (postTxEp wSrc)
+                Default payload
+            expectResponseCode @IO HTTP.status202 rTrans
+
+            rGet <- request @ApiWallet ctx (getWalletEp wDest) Default Empty
+            let coinsSent = map fromIntegral $ take alreadyAbsorbed coins
+            verify rGet
+                [ expectEventually ctx balanceTotal (fromIntegral $ sum coinsSent)
+                , expectEventually ctx balanceAvailable (fromIntegral $ sum coinsSent)
+                ]
+            --verify utxo
+            rStat1 <- request @ApiUtxoStatistics ctx (getWalletUtxoEp wDest) Default Empty
+            expectResponseCode @IO HTTP.status200 rStat1
+            expectWalletUTxO coinsSent (snd rStat1)
+
+    describe "WALLETS_UTXO_03 - non-existing wallets" $  do
+        forM_ falseWalletIds $ \(title, walId) -> it title $ \ctx -> do
+            let endpoint =
+                        "v2/wallets/"
+                        <> T.pack walId
+                        <> "/statistics/utxos" :: Text
```

Comment:
What about adding a `getUtxoStatisticsEp` in the DSL next to the other endpoints helpers ?

Reply:
Yes there is such endpoint `getWalletUtxoEp :: ApiWallet -> (Method, Text)`, but, as others, it takes `ApiWallet` as an argument. Here we only have `walId` and there is yet another place in the test code like that, so I, kind of, didn't bother adding `getWalletUtxoEp' :: Text -> (Method, Text)`.